### ISSUE TITLE
MSS620 FW:2.1.7, HW:2.0.0 Stopped responding in HA

### ISSUE THREAD

**Issue by CTJohnK** [krahabb/meross_lan#159]
One of my two MSS620 outdoor plugs stopped responding in HA yesterday.  I had time to research today and I'm not seeing an obvious reason.

- Meross LAN v2.4.3
- The device responds to the Meross app.
- The device responds to Google Assistant (expected since the app works).
- I can ping the device on the network.
- I can ping the device from HA Terminal.
- I have NOT updated the firmware and confirmed the other unit (which is responding) is running the same FW and HW versions.
- I deleted the unit from Meross LAN.
- 'Add Integration', selected 'Meross LAN', entered the IP address (key was already populated), pressed 'Submit' and received the error "Unable to connect."
- I tried entering '2222' for the key, clicked Submit twice, entered my Meross credentials and was brought back to the "Unable to connect." error message.
- I tried using a non-existent IP address as well and received the same error message, as expected.  This leads me to believe Meross LAN can't find the device, but I do not understand why.
- I then selected one of the other devices, Configuration, and enabled debugging.  I tried the above steps to add the device.  After disabling debugging I checked the log and there is nothing about the device I'm trying to add.

Any ideas why Meross LAN cannot 'see' the device while everything else can?

Jan. 21 update: I've restarted HA a few times and tried to add the device again this morning, however I'm still receiving the same error message.

**Comment by Cippo2019** [comment]
HI CTJohnK 
one question (need by me) ..  I cannot help you on that problem
when you wirte : "I deleted the unit from Meross LAN" .. how you did that ?
I want to delete one device but I dont understand how.

thanks a lot 
Tiziano

**Comment by CTJohnK** [comment]
Tiziano, from the Integration screen, click the ">" of the Meross LAN device you want to remove.  Now click on the three vertical dots (hamburger) icon and select "Delete" or "Disable".

**Comment by Cippo2019** [comment]
> Tiziano, from the Integration screen, click the ">" of the Meross LAN device you want to remove. Now click on the three vertical dots (hamburger) icon and select "Delete" or "Disable".

a ok... I thought that this option would have canceled all integration and not only the selected one.
thanks a lot. 
:-)

**Comment by wsw70** [comment]
> I thought that this option would have canceled all integration and not only the selected one.

This is a common concern :) Pretty much everyone went through the "will this remove the whole integration" question. I think the name of the action is not very fortunate.

**Comment by CTJohnK** [comment]
Just an update, I did try again today with the same results, but I did turn on the debug tracing, rebooted HA and the result from the log is below.

2022-01-26 17:04:45 DEBUG (MainThread) [custom_components.meross_lan] MerossHttpClient(192.168.20.171): HTTP POST method:(GET) namespace:(Appliance.System.All)
2022-01-26 17:04:46 DEBUG (MainThread) [custom_components.meross_lan] MerossHttpClient(192.168.20.171): HTTP Exception (Server disconnected)
2022-01-26 17:04:46 WARNING (MainThread) [custom_components.meross_lan] Error (Server disconnected) configuring meross device (host:192.168.20.171)

**Comment by krahabb** [comment]
Hello @CTJohnK ,
I have no idea too. 
It happened sometimes in user issues but I've always regarded this 'server disconnected' behaviour as a possibly wrong key with a possibly 'intolerant' firmware.
This is not the case though since they key is 'pretty' correct and the device too was working correctly with its current firmware.
I guess you've already tried to reboot the device too.

**Comment by CTJohnK** [comment]
Well, this is a mystery.  A short time ago, I went into the HA UI and there was a notification.  I checked the notifications and it said a new device had been discovered.  Well, you might have guessed, it's my missing MSS620.  I accepted it and it is now working again.

I have no idea what changed.  I have not upgraded to the latest HA code yet, the system has been stopped/started a number of times while trying to figure out why the unit stopped working in HA and many times since I reported the issue.  I did fail to mention I had unplugged/re-plugged the unit a couple of times during my initial diagnostics.  We have not lost power either.
